glide/build.gradle

```diff
@@ -93,42 +95,40 @@ def getAndroidJar() {
     "${getAndroidSdkDirectory()}/platforms/${getAndroidCompileSdkVersion()}/android.jar"
 }
 
-def getSupportJar() {
-    "${getAndroidSdkDirectory()}/extras/android/support/v4/android-support-v4.jar"
-}
-
 project.archivesBaseName = "${POM_ARTIFACT_ID}-${VERSION_NAME}"
 
 // Generate javadocs and sources containing batched documentation and sources for all internal
 // projects.
 ['release', 'debug'].each { variantName ->
 
     task("${variantName}SourceJar", type: Jar) {
-      from getSourceFilesForVariantJar(variantName)
+        from getSourceFilesForVariantJar(variantName)
     }
 
-    def javadocTask = task("${variantName}Javadoc", type: Javadoc) {
+    def javadocTask = tasks.create("${variantName}Javadoc", Javadoc) {
         source = getSourceFilesForVariantJar(variantName)
         source += getSourceFilesForVariantJavadoc(variantName)
 
-        def classpathFiles = files(getAndroidLibraryVariantsForJar(variantName).collect {
-            files(it.javaCompile.classpath.files, getAndroidJar(), getSupportJar())
-        })
-        classpathFiles += files(getAndroidLibraryVariantsForJavadoc(variantName).collect {
-            files(it.javaCompile.classpath.files, getAndroidJar(), getSupportJar())
-        })
-        classpathFiles += (getInternalJavaProjectsForJavadoc().collect {
-            it.sourceSets.main.compileClasspath.files
-        }.flatten())
-        classpathFiles += (getInternalJavaProjectsForJar().collect {
-            it.sourceSets.main.compileClasspath.files
-        }.flatten())
-        // Finds dependencies of Android packages that would otherwise be ignored (Volley in
-        // particular)
-        classpathFiles += getInternalAndroidProjectsForJavadoc().collect {
-            files("${it.projectDir}/build/intermediates/classes/release")
-        }
-        classpath = files(classpathFiles)
+        classpath = project.files(
```

Comment:
Missing getSupportJar here too, maybe intentional?

Reply:
Intentional, notice that the whole method was removed 20 lines above.

It should be on classpath of `glide/library`, like any other dependency. Probably a residue from the times when the m2 repo was not in the SDK.

Tested these changes by diffing `println classpath.files*.absolutePath.join('\n')` before and after.

For me `C:\Users\TWiStEr\.android\build-cache\5377d27e8ef13115d2f05979b4cb19d3e684944b\output\jars\classes.jar` was flagged as input from `com.android.support\support-v4\26.0.2\762902b1aa5eb48bdb1a1f84b1be19a7b1285708\support-v4-26.0.2.aar` and the other split support lib was also found similarly with other build-cache hashes.